commands/chain.go

```diff
@@ -40,37 +45,177 @@ var ChainCmd = &cli.Command{
 		ChainListCmd,
 		ChainSetHeadCmd,
 		ChainActorCodesCmd,
+		ChainActorMethodsCmd,
 		ChainStateInspect,
 		ChainStateCompute,
 		ChainStateComputeRange,
 	},
 }
 
+type chainActorOpts struct {
+	persist bool
+	config  string
+	storage string
+}
+
+var chainActorFlags chainActorOpts
+
+var persistFlag = &cli.BoolFlag{
+	Name:        "persist",
+	Value:       false,
+	Usage:       "Whether to persist the data to the backing storage.",
+	Destination: &chainActorFlags.persist,
+}
+
+var configFlag = &cli.StringFlag{
+	Name:        "config",
+	Usage:       "Specify path of config file to use.",
+	EnvVars:     []string{"LILY_CONFIG"},
+	Destination: &chainActorFlags.config,
+}
+
+var storageFlag = &cli.StringFlag{
+	Name:        "storage",
+	Usage:       "Specify the storage to use, if persisting the displayed output.",
+	Destination: &chainActorFlags.storage,
+}
+
 var ChainActorCodesCmd = &cli.Command{
 	Name:  "actor-codes",
-	Usage: "Print actor codes and names",
+	Usage: "Print actor codes and names.",
+	Flags: []cli.Flag{persistFlag, configFlag, storageFlag},
 	Action: func(cctx *cli.Context) error {
-		for _, a := range []string{manifest.AccountKey, manifest.CronKey, manifest.DatacapKey, manifest.InitKey, manifest.MarketKey, manifest.MinerKey, manifest.MultisigKey, manifest.PaychKey, manifest.PowerKey, manifest.RewardKey, manifest.SystemKey, manifest.VerifregKey} {
+		manifests := manifest.GetBuiltinActorsKeys(actorstypes.Version10)
+		t := table.NewWriter()
+		t.AppendHeader(table.Row{"name", "family", "code"})
+
+		// values that may be accessed if user wants to persist to Storage
+		var results common.ActorCodeList
+		var strg model.Storage
+		var ctx context.Context
+
+		if chainActorFlags.persist {
+			results = common.ActorCodeList{}
+
+			cfg, err := config.FromFile(chainActorFlags.config)
+			if err != nil {
+				return err
+			}
+
+			md := storage.Metadata{
+				JobName: chainActorFlags.storage,
+			}
+
+			// context for db connection
+			ctx = context.Background()
+
+			sc, err := storage.NewCatalog(cfg.Storage)
+			if err != nil {
+				return err
+			}
+			strg, err = sc.Connect(ctx, chainActorFlags.storage, md)
+			if err != nil {
+				return err
+			}
+		}
+
+		for _, a := range manifests {
 			av := make(map[actorstypes.Version]cid.Cid)
-			for _, v := range []int{0, 2, 3, 4, 5, 6, 7, 8, 9} {
```

Comment:
make this list a global in the file for easier updating of these methods when new actor versions are released.